src/lib/stores/walletStore.js

```diff
@@ -16,11 +16,52 @@ import { TransactionBuilder } from 'stellar-sdk'
 
 function createWalletStore() {
     /** @type {import('svelte/store').Writable<WalletStore>} */
-    const { subscribe, set } = persisted('bpa:walletStore', { keyId: '', publicKey: '' })
-
+    const { subscribe, set, } = persisted('bpa:walletStore', { keyId: '', publicKey: '' })
+    const store = persisted('bpa:walletStore', { keyId: '', publicKey: '' })
     return {
         subscribe,
 
+         /**
+         * Checks if a wallet is already registered
+         * @param {string} publicKey Public Stellar address
+         * @returns {Promise<boolean>} True if the wallet is registered, false otherwise
+         */
+         isRegistered: async (publicKey) => {
+            const storeValue = get(store)
+            return storeValue.publicKey === publicKey
+        },
+
+
+        /**
+         * Registers a user by their public key (wallet-based registration)
+         * @param {Object} opts Options object
+         * @param {string} opts.publicKey Public Stellar address
+         */
+        registerWithWallet: async ({ publicKey }) => {
+            try {
+                store.set({
+                    keyId: publicKey, // Using publicKey as keyId for simplicity
+                    publicKey: publicKey,
+                })
+            } catch (err) {
+                console.error('Error registering wallet', err)
+                // throw new Error(err.toString())
+            }
+        },
+
+        /**
+         * Logs in a user by their public key (wallet-based login)
+         * @param {string} publicKey Public Stellar address
+         */
+        loginWithWallet: async (publicKey) => {
+            const storeValue = get(store)
+            if (storeValue.publicKey !== publicKey) {
+                throw new Error('Wallet not registered')
+            }
```

Comment:
```suggestion
            if (get(walletStore).publicKey !== publicKey) {
                throw error(400, { message: 'Wallet not registered' })
            }
```
Let's use the same error throwing techniques that are used in the other functions.

Reply:
replaced the error throwing technique to fit the previous one